src/src/WebServer/Metrics.cpp

```diff
@@ -0,0 +1,190 @@
+#include "../Helpers/StringGenerator_GPIO.h"
+#include "../Helpers/StringGenerator_System.h"
+
+#include "../Static/WebStaticData.h"
+
+#ifdef USES_MQTT
+# include "../Globals/MQTT.h"
+# include "../Helpers/PeriodicalActions.h" // For finding enabled MQTT controller
+#endif
+
+#ifdef ESP32
+# include <esp_partition.h>
+#endif // ifdef ESP32
+
+
+void handle_metrics() {
+    int result;
+    String resultString;
+    resultString.reserve(1000);
+
+
+    //uptime
+    resultString = F("# HELP espeasy_uptime current device uptime in minutes\n");
+    resultString += F("# TYPE espeasy_uptime counter\n");        
+    resultString += "espeasy_uptime ";
+    resultString += getValue(LabelType::UPTIME);       
+    resultString += "\n";
+  
+
+    //load
+    resultString += F("# HELP espeasy_load device percentage load\n");
+    resultString += F("# TYPE espeasy_load gauge\n");         
+    resultString += "espeasy_load ";
+    resultString += getValue(LabelType::LOAD_PCT);     
+    resultString += "\n";
+
+    //Free RAM
+    resultString += F("# HELP espeasy_free_ram device amount of RAM free in Bytes\n");
+    resultString += F("# TYPE espeasy_free_ram gauge\n");    
+    resultString += "espeasy_free_ram ";
+    resultString += getValue(LabelType::FREE_MEM);   
+    resultString += "\n";
+
+    //Free RAM
+    resultString += F("# HELP espeasy_free_stack device amount of Stack free in Bytes\n");
+    resultString += F("# TYPE espeasy_free_stack gauge\n");     
+    resultString += "espeasy_free_stack ";
+    resultString += getValue(LabelType::FREE_STACK);  
+    resultString += "\n";
+
+    //Wifi strength
+    resultString += F("# HELP espeasy_wifi_rssi Wifi connection Strength\n");
+    resultString += F("# TYPE espeasy_wifi_rssi gauge\n");   
+    resultString += "espeasy_wifi_rssi ";
+    resultString += getValue(LabelType::WIFI_RSSI); 
+    resultString += "\n";
+
+    //Wifi uptime
+    resultString += F("# HELP espeasy_wifi_connected Time wifi has been connected in milliseconds\n");
+    resultString += F("# TYPE espeasy_wifi_connected counter\n");    
+    resultString += "espeasy_wifi_connected ";
+    resultString += getValue(LabelType::CONNECTED_MSEC);
+    resultString += "\n";
+
+    //Wifi reconnects
+    resultString += F("# HELP espeasy_wifi_reconnects Number of times Wifi has reconnected since boot\n");
+    resultString += F("# TYPE espeasy_wifi_reconnects counter\n");   
+    resultString += "espeasy_wifi_reconnects ";
+    resultString += getValue(LabelType::NUMBER_RECONNECTS); 
+    resultString += "\n";
+
+    //devices
+    resultString += handle_metrics_devices();
+
+
+
+    web_server.send(200, F("text/plain;"), resultString);
+
+}
+
+
+
+String handle_metrics_value_name(const String& valName){
+    return String(valName);
+}
+
+String handle_metrics_value_value(const String& valValue){
+    return String(valValue);
+}
+
+
+
+String handle_metrics_devices(){
+    String returnString = "";
```

Comment:
No need to initialize with an empty string.
The default constructor of `String` does create a valid empty string.

Reply:
This one still appears to be present.
`String returnString;` is enough.
May sound strange, but adding an empty string initialization like this does really add unneeded code, including using extra RAM as it is a string and not a F() macro string.
And like I said, the default constructor of `String` is a valid empty string.